### PR TITLE
Table widget rendering bug

### LINKED ISSUE #69395: Table cell content does not render if FutureBuilder returns widget of different type when state changes
When using a `FutureBuilder` inside of a `Table`, if the `FutureBuilder` returns different widget type, the second widget does not get rendered. For example, when rendering `CircularProgressIndicator` while `snapshot.hasData` is `false` then switching to `Text` when `snapshot.hasData` is `true` will result in empty cells. Flutter Inspector shows the state of those widgets as `NEEDS-LAYOUT NEEDS-PAINT DETACHED`. The widgets stay in this state until Flutter is asked to re-render the widgets. 

In `Flutter 1.12.13+hotfix.4` this works as expected, with the table content updated to the correct widget when the Future state changes.

## Steps to Reproduce
Here's a sample code that reproduces the bug:
```dart
import 'package:flutter/material.dart';

void main() {
  runApp(MyApp());
}

class MyApp extends StatefulWidget {
  @override
  _MyAppState createState() => _MyAppState();
}

class _MyAppState extends State<MyApp> {
  Future<int> value;

  @override
  initState() {
    super.initState();
    value = Future.delayed(Duration(seconds: 3), () => 31415);
  }

  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Issue Repro',
      theme: ThemeData(
        primarySwatch: Colors.blue,
      ),
      home: new Scaffold(
        appBar: AppBar(title: Text('Issue Repro')),
        body: Column(
          children: [
            Table(
              defaultVerticalAlignment: TableCellVerticalAlignment.middle,
              columnWidths: {
                0: IntrinsicColumnWidth(),
              },
              children: [
                TableRow(children: [
                  Text('Progress Indicator to Text'),
                  FutureBuilder<int>(
                    future: value,
                    builder: (context, snapshot) {
                      if (snapshot.hasData) {
                        return Text('${snapshot.data}');
                      }
                      return Center(
                        child: SizedBox(
                          width: 14.0,
                          height: 14.0,
                          child: CircularProgressIndicator(
                            strokeWidth: 1.0,
                          ),
                        ),
                      );
                    },
                  )
                ]),
                TableRow(children: [
                  Text('Text to Text'),
                  FutureBuilder<int>(
                    future: value,
                    builder: (context, snapshot) {
                      if (snapshot.hasData) {
                        return Text('${snapshot.data}');
                      }
                      return Text('Loading...');
                    },
                  )
                ]),
                TableRow(children: [
                  Text('Container to Text'),
                  FutureBuilder<int>(
                    future: value,
                    builder: (context, snapshot) {
                      if (snapshot.hasData) {
                        return Text('${snapshot.data}');
                      }
                      return Container(
                        color: Colors.blue.shade600,
                        height: 60.0,
                      );
                    },
                  )
                ]),
                TableRow(children: [
                  Text('Container to Container'),
                  FutureBuilder<int>(
                    future: value,
                    builder: (context, snapshot) {
                      if (snapshot.hasData) {
                        return Container(
                          color: Colors.green.shade600,
                          height: 48.0,
                        );
                      }
                      return Container(
                        color: Colors.red.shade600,
                        height: 24.0,
                      );
                    },
                  )
                ]),
              ],
            ),
          ],
        ),
      ),
    );
  }
}
```

**Expected results:**
Table should render correct content returned by the FutureBuilder.

**Actual results:**
Before future is resolved:
Screenshot_1604058883
After future is resolved:
Screenshot_1604058902
Flutter Inspector Output:
image

<details>
  <summary>Logs</summary>

## Flutter Run --verbose Logs
```
[ +141 ms] executing: [C:\DevTools\flutter/] git -c log.showSignature=false log -n 1 --pretty=format:%H
[  +62 ms] Exit code 0 from: git -c log.showSignature=false log -n 1 --pretty=format:%H
[        ] 84f3d28555368a70270e9ac8390a9441df95e752
[        ] executing: [C:\DevTools\flutter/] git tag --points-at HEAD
[  +73 ms] Exit code 0 from: git tag --points-at HEAD
[        ] 1.22.2
[  +10 ms] executing: [C:\DevTools\flutter/] git rev-parse --abbrev-ref --symbolic @{u}
[  +44 ms] Exit code 0 from: git rev-parse --abbrev-ref --symbolic @{u}
[        ] origin/stable
[        ] executing: [C:\DevTools\flutter/] git ls-remote --get-url origin
[  +30 ms] Exit code 0 from: git ls-remote --get-url origin
[        ] https://github.com/flutter/flutter.git
[  +73 ms] executing: [C:\DevTools\flutter/] git rev-parse --abbrev-ref HEAD
[  +51 ms] Exit code 0 from: git rev-parse --abbrev-ref HEAD
[        ] stable
[ +402 ms] Artifact Instance of 'AndroidMavenArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'AndroidGenSnapshotArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'AndroidInternalBuildArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'IOSEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'FlutterWebSdk' is not required, skipping update.
[   +2 ms] Artifact Instance of 'WindowsEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'MacOSEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'LinuxEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'LinuxFuchsiaSDKArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'MacOSFuchsiaSDKArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'FlutterRunnerSDKArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'FlutterRunnerDebugSymbols' is not required, skipping update.
[  +21 ms] executing: C:\DevTools\Android\Sdk\platform-tools\adb.exe devices -l
[  +43 ms] List of devices attached
           emulator-5554          device product:sdk_gphone_x86_arm model:AOSP_on_IA_Emulator device:generic_x86_arm transport_id:2
[   +5 ms] C:\DevTools\Android\Sdk\platform-tools\adb.exe -s emulator-5554 shell getprop
[  +54 ms] Artifact Instance of 'AndroidMavenArtifacts' is not required, skipping update.
[   +3 ms] Artifact Instance of 'AndroidInternalBuildArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'IOSEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'FlutterWebSdk' is not required, skipping update.
[   +1 ms] Artifact Instance of 'WindowsEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'MacOSEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'LinuxEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'LinuxFuchsiaSDKArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'MacOSFuchsiaSDKArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'FlutterRunnerSDKArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'FlutterRunnerDebugSymbols' is not required, skipping update.
[ +277 ms] Generating C:\Users\ashio\Git\Scratchpad\flutter_table_rendering_bug\android\app\src\main\java\io\flutter\plugins\GeneratedPluginRegistrant.java
[  +31 ms] ro.hardware = ranchu
[  +39 ms] Starting incremental build...
[   +1 ms] Initializing file store
[  +19 ms] Skipping target: gen_localizations
[   +5 ms] complete
[   +3 ms] Launching lib\main.dart on AOSP on IA Emulator in debug mode...
[   +5 ms] C:\DevTools\flutter\bin\cache\dart-sdk\bin\dart.exe --disable-dart-dev C:\DevTools\flutter\bin\cache\artifacts\engine\windows-x64\frontend_server.dart.snapshot --sdk-root C:\DevTools\flutter\bin\cache\artifacts\engine\common\flutter_patched_sdk/ --incremental --target=flutter --debugger-module-names --experimental-emit-debug-metadata -Ddart.developer.causal_async_stacks=true --output-dill C:\Users\ashio\AppData\Local\Temp\flutter_tools.84985777\flutter_tool.a52ebedc\app.dill --packages .packages -Ddart.vm.profile=false -Ddart.vm.product=false --bytecode-options=source-positions,local-var-info,debugger-stops,instance-field-initializers,keep-unreachable-code,avoid-closure-call-instructions --enable-asserts --track-widget-creation --initialize-from-dill build\cache.dill.track.dill
[   +7 ms] executing: C:\DevTools\Android\Sdk\build-tools\29.0.2\aapt dump xmltree C:\Users\ashio\Git\Scratchpad\flutter_table_rendering_bug\build\app\outputs\flutter-apk\app.apk AndroidManifest.xml
[  +14 ms] Exit code 0 from: C:\DevTools\Android\Sdk\build-tools\29.0.2\aapt dump xmltree C:\Users\ashio\Git\Scratchpad\flutter_table_rendering_bug\build\app\outputs\flutter-apk\app.apk AndroidManifest.xml
[        ] N: android=http://schemas.android.com/apk/res/android
             E: manifest (line=2)
               A: android:versionCode(0x0101021b)=(type 0x10)0x1
               A: android:versionName(0x0101021c)="1.0.0" (Raw: "1.0.0")
               A: android:compileSdkVersion(0x01010572)=(type 0x10)0x1d
               A: android:compileSdkVersionCodename(0x01010573)="10" (Raw: "10")
               A: package="com.example.flutter_table_rendering_bug" (Raw: "com.example.flutter_table_rendering_bug")
               A: platformBuildVersionCode=(type 0x10)0x1d
               A: platformBuildVersionName=(type 0x10)0xa
               E: uses-sdk (line=7)
                 A: android:minSdkVersion(0x0101020c)=(type 0x10)0x10
                 A: android:targetSdkVersion(0x01010270)=(type 0x10)0x1d
               E: uses-permission (line=14)
                 A: android:name(0x01010003)="android.permission.INTERNET" (Raw: "android.permission.INTERNET")
               E: application (line=22)
                 A: android:label(0x01010001)="flutter_table_rendering_bug" (Raw: "flutter_table_rendering_bug")
                 A: android:icon(0x01010002)=@0x7f080000
                 A: android:name(0x01010003)="io.flutter.app.FlutterApplication" (Raw: "io.flutter.app.FlutterApplication")
                 A: android:debuggable(0x0101000f)=(type 0x12)0xffffffff
                 A: android:appComponentFactory(0x0101057a)="androidx.core.app.CoreComponentFactory" (Raw: "androidx.core.app.CoreComponentFactory")
                 E: activity (line=28)
                   A: android:theme(0x01010000)=@0x7f0a0000
                   A: android:name(0x01010003)="com.example.flutter_table_rendering_bug.MainActivity" (Raw: "com.example.flutter_table_rendering_bug.MainActivity")
                   A: android:launchMode(0x0101001d)=(type 0x10)0x1
                   A: android:configChanges(0x0101001f)=(type 0x11)0x40003fb4
                   A: android:windowSoftInputMode(0x0101022b)=(type 0x11)0x10
                   A: android:hardwareAccelerated(0x010102d3)=(type 0x12)0xffffffff
                   E: meta-data (line=42)
                     A: android:name(0x01010003)="io.flutter.embedding.android.NormalTheme" (Raw: "io.flutter.embedding.android.NormalTheme")
                     A: android:resource(0x01010025)=@0x7f0a0001
                   E: meta-data (line=52)
                     A: android:name(0x01010003)="io.flutter.embedding.android.SplashScreenDrawable" (Raw: "io.flutter.embedding.android.SplashScreenDrawable")
                     A: android:resource(0x01010025)=@0x7f040000
                   E: intent-filter (line=56)
                     E: action (line=57)
                       A: android:name(0x01010003)="android.intent.action.MAIN" (Raw: "android.intent.action.MAIN")
                     E: category (line=59)
                       A: android:name(0x01010003)="android.intent.category.LAUNCHER" (Raw: "android.intent.category.LAUNCHER")
                 E: meta-data (line=66)
                   A: android:name(0x01010003)="flutterEmbedding" (Raw: "flutterEmbedding")
                   A: android:value(0x01010024)=(type 0x10)0x2
[   +6 ms] executing: C:\DevTools\Android\Sdk\platform-tools\adb.exe -s emulator-5554 shell -x logcat -v time -t 1
[  +43 ms] Exit code 0 from: C:\DevTools\Android\Sdk\platform-tools\adb.exe -s emulator-5554 shell -x logcat -v time -t 1
[        ] --------- beginning of main
           10-30 20:26:59.029 I/GnssLocationProvider( 1902): WakeLock released by handleMessage(REPORT_SV_STATUS, 0, com.android.server.location.GnssLocationProvider$SvStatusInfo@af1c77e)
[   +1 ms] executing: C:\DevTools\Android\Sdk\platform-tools\adb.exe -s emulator-5554 shell -x logcat -v time -t 1
[  +34 ms] Exit code 0 from: C:\DevTools\Android\Sdk\platform-tools\adb.exe -s emulator-5554 shell -x logcat -v time -t 1
[        ] --------- beginning of main
           10-30 20:26:59.029 I/GnssLocationProvider( 1902): WakeLock released by handleMessage(REPORT_SV_STATUS, 0, com.android.server.location.GnssLocationProvider$SvStatusInfo@af1c77e)
[   +4 ms] <- compile package:flutter_table_rendering_bug/main.dart
[  +49 ms] executing: C:\DevTools\Android\Sdk\platform-tools\adb.exe version
[ +124 ms] Android Debug Bridge version 1.0.41
           Version 30.0.4-6686687
           Installed as C:\DevTools\Android\Sdk\platform-tools\adb.exe
[  +12 ms] executing: C:\DevTools\Android\Sdk\platform-tools\adb.exe start-server
[ +115 ms] Building APK
Running Gradle task 'assembleDebug'...
[ +128 ms] gradle.properties already sets `android.enableR8`
[  +18 ms] Using gradle from C:\Users\ashio\Git\Scratchpad\flutter_table_rendering_bug\android\gradlew.bat.
[   +3 ms] C:\Users\ashio\Git\Scratchpad\flutter_table_rendering_bug\android\gradlew.bat mode: 33279 rwxrwxrwx.
[  +53 ms] executing: C:\Users\ashio\AppData\Local\JetBrains\Toolbox\apps\AndroidStudio\ch-0\183.5522156\jre\bin\java -version
[ +496 ms] Exit code 0 from: C:\Users\ashio\AppData\Local\JetBrains\Toolbox\apps\AndroidStudio\ch-0\183.5522156\jre\bin\java -version
[   +1 ms] openjdk version "1.8.0_152-release"
           OpenJDK Runtime Environment (build 1.8.0_152-release-1343-b01)
           OpenJDK 64-Bit Server VM (build 25.152-b01, mixed mode)
[   +9 ms] executing: C:\Users\ashio\AppData\Local\JetBrains\Toolbox\apps\AndroidStudio\ch-0\191.5791312\jre\bin\java -version
[ +517 ms] Exit code 0 from: C:\Users\ashio\AppData\Local\JetBrains\Toolbox\apps\AndroidStudio\ch-0\191.5791312\jre\bin\java -version
[        ] openjdk version "1.8.0_202-release"
           OpenJDK Runtime Environment (build 1.8.0_202-release-1483-b03)
           OpenJDK 64-Bit Server VM (build 25.202-b03, mixed mode)
[   +3 ms] executing: [C:\Users\ashio\Git\Scratchpad\flutter_table_rendering_bug\android/] C:\Users\ashio\Git\Scratchpad\flutter_table_rendering_bug\android\gradlew.bat -q -Ptarget-platform=android-x86 -Ptarget=C:\Users\ashio\Git\Scratchpad\flutter_table_rendering_bug\lib\main.dart -Ptrack-widget-creation=true -Pfilesystem-scheme=org-dartlang-root assembleDebug
[+25695 ms] calculateSha: LocalDirectory: 'C:\Users\ashio\Git\Scratchpad\flutter_table_rendering_bug\build\app\outputs\flutter-apk'/app.apk
[  +16 ms] calculateSha: reading file took 16us
[ +485 ms] calculateSha: computing sha took 485us
[   +7 ms] √ Built build\app\outputs\flutter-apk\app-debug.apk.
[   +3 ms] executing: C:\DevTools\Android\Sdk\build-tools\29.0.2\aapt dump xmltree C:\Users\ashio\Git\Scratchpad\flutter_table_rendering_bug\build\app\outputs\flutter-apk\app.apk AndroidManifest.xml
[  +26 ms] Exit code 0 from: C:\DevTools\Android\Sdk\build-tools\29.0.2\aapt dump xmltree C:\Users\ashio\Git\Scratchpad\flutter_table_rendering_bug\build\app\outputs\flutter-apk\app.apk AndroidManifest.xml
[        ] N: android=http://schemas.android.com/apk/res/android
             E: manifest (line=2)
               A: android:versionCode(0x0101021b)=(type 0x10)0x1
               A: android:versionName(0x0101021c)="1.0.0" (Raw: "1.0.0")
               A: android:compileSdkVersion(0x01010572)=(type 0x10)0x1d
               A: android:compileSdkVersionCodename(0x01010573)="10" (Raw: "10")
               A: package="com.example.flutter_table_rendering_bug" (Raw: "com.example.flutter_table_rendering_bug")
               A: platformBuildVersionCode=(type 0x10)0x1d
               A: platformBuildVersionName=(type 0x10)0xa
               E: uses-sdk (line=7)
                 A: android:minSdkVersion(0x0101020c)=(type 0x10)0x10
                 A: android:targetSdkVersion(0x01010270)=(type 0x10)0x1d
               E: uses-permission (line=14)
                 A: android:name(0x01010003)="android.permission.INTERNET" (Raw: "android.permission.INTERNET")
               E: application (line=22)
                 A: android:label(0x01010001)="flutter_table_rendering_bug" (Raw: "flutter_table_rendering_bug")
                 A: android:icon(0x01010002)=@0x7f080000
                 A: android:name(0x01010003)="io.flutter.app.FlutterApplication" (Raw: "io.flutter.app.FlutterApplication")
                 A: android:debuggable(0x0101000f)=(type 0x12)0xffffffff
                 A: android:appComponentFactory(0x0101057a)="androidx.core.app.CoreComponentFactory" (Raw: "androidx.core.app.CoreComponentFactory")
                 E: activity (line=28)
                   A: android:theme(0x01010000)=@0x7f0a0000
                   A: android:name(0x01010003)="com.example.flutter_table_rendering_bug.MainActivity" (Raw: "com.example.flutter_table_rendering_bug.MainActivity")
                   A: android:launchMode(0x0101001d)=(type 0x10)0x1
                   A: android:configChanges(0x0101001f)=(type 0x11)0x40003fb4
                   A: android:windowSoftInputMode(0x0101022b)=(type 0x11)0x10
                   A: android:hardwareAccelerated(0x010102d3)=(type 0x12)0xffffffff
                   E: meta-data (line=42)
                     A: android:name(0x01010003)="io.flutter.embedding.android.NormalTheme" (Raw: "io.flutter.embedding.android.NormalTheme")
                     A: android:resource(0x01010025)=@0x7f0a0001
                   E: meta-data (line=52)
                     A: android:name(0x01010003)="io.flutter.embedding.android.SplashScreenDrawable" (Raw: "io.flutter.embedding.android.SplashScreenDrawable")
                     A: android:resource(0x01010025)=@0x7f040000
                   E: intent-filter (line=56)
                     E: action (line=57)
                       A: android:name(0x01010003)="android.intent.action.MAIN" (Raw: "android.intent.action.MAIN")
                     E: category (line=59)
                       A: android:name(0x01010003)="android.intent.category.LAUNCHER" (Raw: "android.intent.category.LAUNCHER")
                 E: meta-data (line=66)
                   A: android:name(0x01010003)="flutterEmbedding" (Raw: "flutterEmbedding")
                   A: android:value(0x01010024)=(type 0x10)0x2
[   +1 ms] Stopping app 'app.apk' on AOSP on IA Emulator.
[        ] executing: C:\DevTools\Android\Sdk\platform-tools\adb.exe -s emulator-5554 shell am force-stop com.example.flutter_table_rendering_bug
[  +53 ms] executing: C:\DevTools\Android\Sdk\platform-tools\adb.exe -s emulator-5554 shell pm list packages com.example.flutter_table_rendering_bug
[ +180 ms] package:com.example.flutter_table_rendering_bug
[   +8 ms] executing: C:\DevTools\Android\Sdk\platform-tools\adb.exe -s emulator-5554 shell cat /data/local/tmp/sky.com.example.flutter_table_rendering_bug.sha1
[ +183 ms] e685ef6edc1790a34cf43eebe5146759841b13b7
[   +3 ms] Installing APK.
[   +6 ms] executing: C:\DevTools\Android\Sdk\platform-tools\adb.exe version
[  +42 ms] Android Debug Bridge version 1.0.41
           Version 30.0.4-6686687
           Installed as C:\DevTools\Android\Sdk\platform-tools\adb.exe
[        ] executing: C:\DevTools\Android\Sdk\platform-tools\adb.exe start-server
Installing build\app\outputs\flutter-apk\app.apk...
[  +30 ms] executing: C:\DevTools\Android\Sdk\platform-tools\adb.exe -s emulator-5554 install -t -r C:\Users\ashio\Git\Scratchpad\flutter_table_rendering_bug\build\app\outputs\flutter-apk\app.apk
[+4083 ms] Performing Streamed Install
           Success
[   +7 ms] executing: C:\DevTools\Android\Sdk\platform-tools\adb.exe -s emulator-5554 shell echo -n d94e0e0d384a018da996202e893878f2c8a209c5 > /data/local/tmp/sky.com.example.flutter_table_rendering_bug.sha1
[ +118 ms] AOSP on IA Emulator startApp
[   +5 ms] executing: C:\DevTools\Android\Sdk\platform-tools\adb.exe -s emulator-5554 shell am start -a android.intent.action.RUN -f 0x20000000 --ez enable-background-compilation true --ez enable-dart-profiling true --ez enable-checked-mode true --ez verify-entry-points true com.example.flutter_table_rendering_bug/com.example.flutter_table_rendering_bug.MainActivity
[ +866 ms] Starting: Intent { act=android.intent.action.RUN flg=0x20000000 cmp=com.example.flutter_table_rendering_bug/.MainActivity (has extras) }
[        ] Waiting for observatory port to be available...
[+6065 ms] Observatory URL on device: http://127.0.0.1:36070/KZyw3KdmbTY=/
[   +9 ms] executing: C:\DevTools\Android\Sdk\platform-tools\adb.exe -s emulator-5554 forward tcp:0 tcp:36070
[ +110 ms] 53512
[   +3 ms] Forwarded host port 53512 to device port 36070 for Observatory
[  +26 ms] Caching compiled dill
[  +84 ms] Connecting to service protocol: http://127.0.0.1:53512/KZyw3KdmbTY=/
[   +1 ms] DDS is currently disabled due to https://github.com/flutter/flutter/issues/62507
[ +701 ms] Successfully connected to service protocol: http://127.0.0.1:53512/KZyw3KdmbTY=/
Waiting for AOSP on IA Emulator to report its views...
[  +60 ms] DevFS: Creating new filesystem on the device (null)
[  +56 ms] DevFS: Created new filesystem on the device (file:///data/user/0/com.example.flutter_table_rendering_bug/code_cache/flutter_table_rendering_bugYHXJHH/flutter_table_rendering_bug/)
[   +1 ms] Updating assets
Syncing files to device AOSP on IA Emulator...
[  +87 ms] Scanning asset files
[   +8 ms] <- reset
[   +1 ms] Compiling dart to kernel with 0 updated files
[   +6 ms] <- recompile package:flutter_table_rendering_bug/main.dart d5fced0d-a956-4dc3-9c67-ca95bc730c7f
[   +1 ms] <- d5fced0d-a956-4dc3-9c67-ca95bc730c7f
[   +8 ms] I/Choreographer(15129): Skipped 70 frames!  The application may be doing too much work on its main thread.
[   +6 ms] D/EGL_emulation(15129): eglMakeCurrent: 0xe7bc0380: ver 3 0 (tinfo 0xdffb2540)
[  +39 ms] D/eglCodecCommon(15129): setVertexArrayObject: set vao to 0 (0) 1 2
[ +155 ms] Updating files
[+1368 ms] DevFS: Sync finished
[   +5 ms] Synced 1.8MB.
[   +6 ms] <- accept
[  +59 ms] Connected to _flutterView/0xdedc9c10.
[+1301 ms] D/EGL_emulation(15129): eglMakeCurrent: 0xe7bc04a0: ver 3 0 (tinfo 0xdff90b50)
[  +81 ms] D/eglCodecCommon(15129): setVertexArrayObject: set vao to 0 (0) 1 0

```

## Flutter Analyze Output
```
Analyzing flutter_table_rendering_bug...
No issues found! (ran in 4.7s)
```

## Flutter Doctor Output
```
[√] Flutter (Channel stable, 1.22.2, on Microsoft Windows [Version 10.0.18362.1082], locale en-US)
    • Flutter version 1.22.2 at C:\DevTools\flutter
    • Framework revision 84f3d28555 (2 weeks ago), 2020-10-15 16:26:19 -0700
    • Engine revision b8752bbfff
    • Dart version 2.10.2


[√] Android toolchain - develop for Android devices (Android SDK version 29.0.2)
    • Android SDK at C:\DevTools\Android\Sdk
    • Platform android-29, build-tools 29.0.2
    • ANDROID_HOME = C:\DevTools\Android\Sdk
    • Java binary at: C:\Users\ashio\AppData\Local\JetBrains\Toolbox\apps\AndroidStudio\ch-0\191.5791312\jre\bin\java
    • Java version OpenJDK Runtime Environment (build 1.8.0_202-release-1483-b03)
    • All Android licenses accepted.

[√] Android Studio (version 3.4)
    • Android Studio at C:\Users\ashio\AppData\Local\JetBrains\Toolbox\apps\AndroidStudio\ch-0\183.5522156
    • Flutter plugin version 33.3.1
    • Dart plugin version 182.5215
    • Java version OpenJDK Runtime Environment (build 1.8.0_152-release-1343-b01)

[√] Android Studio (version 3.5)
    • Android Studio at C:\Users\ashio\AppData\Local\JetBrains\Toolbox\apps\AndroidStudio\ch-0\191.5791312
    • Flutter plugin version 40.2.2
    • Dart plugin version 191.8580
    • Java version OpenJDK Runtime Environment (build 1.8.0_202-release-1483-b03)

[√] IntelliJ IDEA Ultimate Edition (version 2019.2)
    • IntelliJ at C:\Users\ashio\AppData\Local\JetBrains\Toolbox\apps\IDEA-U\ch-0\192.6603.28
    • Flutter plugin version 42.2.2
    • Dart plugin version 192.7761

[√] VS Code (version 1.50.1)
    • VS Code at C:\Users\ashio\AppData\Local\Programs\Microsoft VS Code
    • Flutter extension version 3.15.1

[√] Connected device (1 available)
    • AOSP on IA Emulator (mobile) • emulator-5554 • android-x86 • Android 9 (API 28) (emulator)

• No issues found!
```

</details>

### PR DESCRIPTION
**Pull request by goderbauer**
Fixes #69395

Thanks to @xu-baolin by doing the ground work for this in https://github.com/flutter/flutter/pull/69670 and https://github.com/flutter/flutter/pull/76375.

### CODE REVIEW

File: packages/flutter/lib/src/widgets/framework.dart [hunk 1]
```diff
@@ -5525,14 +5528,21 @@ abstract class RenderObjectElement extends Element {
...
     assert(newWidgets != null);
+    assert(slots == null || newWidgets.length == slots.length);
 
     Element? replaceWithNullIfForgotten(Element child) {
       return forgottenChildren != null && forgottenChildren.contains(child) ? null : child;
     }
 
+    Object? slotFor(int newChildIndex, Element? previousChild) {
+      return slots != null
+          ? slots[newChildIndex]
```
Comment (xu-baolin) [inline packages/flutter/lib/src/widgets/framework.dart:5542]: nit, Indent two spaces

File: packages/flutter/lib/src/widgets/table.dart [hunk 2]
```diff
@@ -348,37 +348,48 @@ class _TableElement extends RenderObjectElement {
...
+    // Once [mount]/[update] are done, the children are getting set all at once
+    // in [_updateRenderObjectChildren].
+    if (!_doingMountOrUpdate) {
+      renderObject.setChild(slot.column, slot.row, child);
+    }
   }
 
   @override
-  void moveRenderObjectChild(RenderObject child, dynamic oldSlot, dynamic newSlot) {
+  void moveRenderObjectChild(RenderBox child, Object? oldSlot, Object? newSlot) {
```
Comment (xu-baolin) [inline packages/flutter/lib/src/widgets/table.dart:386]: The `oldSlot` and `newSlot` are should `_TableSlot` type here.

File: packages/flutter/lib/src/widgets/table.dart [hunk 3]
```diff
@@ -348,37 +348,48 @@ class _TableElement extends RenderObjectElement {
...
+    assert(_doingMountOrUpdate);
+    // Child gets moved at the end of [update] in [_updateRenderObjectChildren].
   }
 
   @override
-  void removeRenderObjectChild(RenderObject child, dynamic slot) {
+  void removeRenderObjectChild(RenderBox child, Object? slot) {
```
Comment (xu-baolin) [inline packages/flutter/lib/src/widgets/table.dart:392]: Same as above, `slot` should be `_TableSlot` type.

File: packages/flutter/lib/src/widgets/table.dart [hunk 4]
```diff
@@ -417,6 +434,7 @@ class _TableElement extends RenderObjectElement {
       updateChildren(oldChildren, const <Widget>[], forgottenChildren: _forgottenChildren);
 
     _children = newChildren;
+    _doingMountOrUpdate = false;
```
Comment (xu-baolin) [inline packages/flutter/lib/src/widgets/table.dart:437]: Why not set this flag at the end of the method, the `super.update(newWidget);` is not called here, Literally, the update has not been completed yet.

File: packages/flutter/lib/src/widgets/table.dart [hunk 5]
```diff
@@ -489,3 +507,30 @@ class TableCell extends ParentDataWidget<TableCellParentData> {
...
+
+  final int column;
+  final int row;
+
+  @override
+  bool operator ==(Object other) {
+    if (other.runtimeType != runtimeType)
+      return false;
+    return other is _TableSlot
+        && column == other.column
```
Comment (xu-baolin) [inline packages/flutter/lib/src/widgets/table.dart:523]: Indent nit.

File: packages/flutter/lib/src/widgets/table.dart [hunk 6]
```diff
@@ -348,37 +348,48 @@ class _TableElement extends RenderObjectElement {
...
-
   List<_TableElementRow> _children = const<_TableElementRow>[];
 
+  bool _doingMountOrUpdate = false;
+
   @override
-  void mount(Element? parent, dynamic newSlot) {
+  void mount(Element? parent, Object? newSlot) {
     super.mount(parent, newSlot);
+    _doingMountOrUpdate = true;
```
Comment (xu-baolin) [inline packages/flutter/lib/src/widgets/table.dart:358]: assert(!_doingMountOrUpdate);

File: packages/flutter/lib/src/widgets/table.dart [hunk 7]
```diff
@@ -387,6 +398,7 @@ class _TableElement extends RenderObjectElement {
 
   @override
   void update(Table newWidget) {
+    _doingMountOrUpdate = true;
```
Comment (xu-baolin) [inline packages/flutter/lib/src/widgets/table.dart:401]: assert(!_doingMountOrUpdate);

File: packages/flutter/lib/src/widgets/table.dart (hunk 5 above)
Comment (goderbauer) [reply packages/flutter/lib/src/widgets/table.dart:523]: Indentation here is correct as per boilerplate in style guide: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#common-boilerplates-for-operator-and-hashcode

Comment (xu-baolin) [reply packages/flutter/lib/src/widgets/table.dart:527]: This is the first time I noticed this rule, thank you for your reminder!

**Review by xu-baolin — APPROVED** [review]
LGTM

**Comment by JoseGeorges8** [comment]
@xu-baolin Hi, I still encounter this issue on Flutter 2.0.4.

If my TableRow's direct children are FutureBuilders then they won't render. If I however change the parent of this FutureBuilder to any other widget then it renders fine!

**Comment by xu-baolin** [comment]
> rect ch

Can you provide flutter doctor -v and a complete minimal reproducible code sample that shows the issue?
Thanks.

**Pull request merged**